### DISCUSSION TITLE
Alert user when there is unsaved data in the master detail dialog page while clicking on back button.

### DISCUSSION THREAD

**Discussion by vmrosariojoe** [serenity-is/Serenity#5407] [Help] [answered]
Hi Team.
I have a master detail dialog and detail records are only in memory until the master dialog save button is clicked.
If the user accidently clicks on the back button then all the data which are in the memory are removed. 
How to alert the user when the user leaves the page or clicks on back button that there are some unsaved record in the page?

![11](https://user-images.githubusercontent.com/22878845/106106337-597cee80-6180-11eb-8b05-4f19f587329a.PNG)

**Comment by erehtolleh** [comment]
There is exact [sample ](https://serenity.is/demo/Northwind/Customer)here.

**Comment by luongnv6** [comment, marked as answer]
https://github.com/serenity-is/Serene/blob/master/Serene/Serene.Core/Modules/Northwind/Customer/CustomerDialog.ts
this is code demo for you

**Reply by vmrosariojoe** [reply]
Thanks a lot. much appreciated.
